lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -502,15 +534,9 @@ def test_filter_query(self, monkeypatch, client, filters, expected):
             "pbench.server.api.resources.datasets_list.Auth.get_current_user_id",
             lambda: DRB_USER_ID,
         )
-        prefix = (
-            "SELECT datasets.access, datasets.id, datasets.name, "
-            "datasets.owner_id, datasets.resource_id, datasets.uploaded "
-            "FROM datasets LEFT OUTER JOIN dataset_metadata ON datasets.id "
-            "= dataset_metadata.dataset_ref WHERE "
-        )
-        query = DatasetsList.filter_query(
-            filters, Database.db_session.query(Dataset).outerjoin(Metadata)
-        )
+        prefix = "SELECT datasets.access, datasets.id, datasets.name, datasets.owner_id, datasets.resource_id, datasets.uploaded FROM datasets LEFT OUTER JOIN dataset_metadata AS dataset_metadata_1 ON dataset_metadata_1.dataset_ref = datasets.id AND dataset_metadata_1.key = 'metalog' LEFT OUTER JOIN dataset_metadata AS dataset_metadata_2 ON dataset_metadata_2.dataset_ref = datasets.id AND dataset_metadata_2.key = 'server' LEFT OUTER JOIN dataset_metadata AS dataset_metadata_3 ON dataset_metadata_3.dataset_ref = datasets.id AND dataset_metadata_3.key = 'global' LEFT OUTER JOIN dataset_metadata AS dataset_metadata_4 ON dataset_metadata_4.dataset_ref = datasets.id AND dataset_metadata_4.key = 'user' AND dataset_metadata_4.user_id = '3' WHERE "
```

Comment:
Should we also test with prefix without the outer join on the user metadata key?